hazelcast/src/test/java/com/hazelcast/sql/impl/type/converter/ConvertersTest.java

```diff
@@ -0,0 +1,1074 @@
+                break;
+
+            case DATE:
+                assertEquals(expected, converter.canConvertToDate());
+
+                break;
+
+            case TIMESTAMP:
+                assertEquals(expected, converter.canConvertToTimestamp());
+
+                break;
+
+            case TIMESTAMP_WITH_TIMEZONE:
+                assertEquals(expected, converter.canConvertToTimestampWithTimezone());
+
+                break;
+
+            case OBJECT:
+                assertEquals(expected, converter.canConvertToObject());
+
+                break;
+        }
+
+        if (!expected) {
+            checkCannotConvert(converter, typeFamily);
+        }
+    }
+
+    private void checkCannotConvert(Converter converter, QueryDataTypeFamily typeFamily) {
+        try {
+            Object val = new Object();
+
+            switch (typeFamily) {
+                case VARCHAR:
+                    converter.asVarchar(val);
+
+                    break;
+
+                case BIT:
+                    converter.asBit(val);
+
+                    break;
+
+                case TINYINT:
+                    converter.asTinyint(val);
+
+                    break;
+
+                case SMALLINT:
+                    converter.asSmallint(val);
+
+                    break;
+
+                case INT:
+                    converter.asInt(val);
+
+                    break;
+
+                case BIGINT:
+                    converter.asBigint(val);
+
+                    break;
+
+                case DECIMAL:
+                    converter.asDecimal(val);
+
+                    break;
+
+                case REAL:
+                    converter.asReal(val);
+
+                    break;
+
+                case DOUBLE:
+                    converter.asDouble(val);
+
+                    break;
+
+                case TIME:
+                    converter.asTime(val);
+
+                    break;
+
+                case DATE:
+                    converter.asDate(val);
+
+                    break;
+
+                case TIMESTAMP:
+                    converter.asTimestamp(val);
+
+                    break;
+
+                case TIMESTAMP_WITH_TIMEZONE:
+                    converter.asTimestampWithTimezone(val);
+
+                    break;
+
+                case OBJECT:
+                    converter.asObject(val);
+
+                    break;
+
+                default:
+                    return;
+            }
+
+            fail("Must fail: " + typeFamily);
+        } catch (HazelcastSqlException e) {
+            assertEquals(SqlErrorCode.DATA_EXCEPTION, e.getCode());
+        }
+    }
+
+    private void checkConverterSelf(Converter converter) {
+        MockConverter mockConverter = new MockConverter();
+
+        converter.convertToSelf(mockConverter, new Object());
+
+        assertEquals(converter.getTypeFamily(), mockConverter.getInvoked());
+    }
+
+    private static final class CustomClass {
+        private int val;
```

Comment:
Minor: make it `final`?

Reply:
This class will go away in the next PR, where more tests are added. Its counterpart might be used in serialization, so I would avoid final here.